### PR TITLE
docs: Document custom field mappings

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Preview: [View the Elasticsearch index template](https://apm-server_11997.docs-preview.app.elstc.co/guide/en/apm/guide/master/custom-index-template.html#custom-index-template-mappings)

## Motivation/summary

Adds more information to [View the Elasticsearch index template](https://www.elastic.co/guide/en/apm/guide/current/custom-index-template.html) including:

- How to add custom **field mappings** and at least one practical example
- How to add **runtime fields** and at least one practical example

~~@bmorelli25 these ☝️  are the topics included in the [initial issue](https://github.com/elastic/observability-docs/issues/2206), but should we also add the following while we're updating this page?~~

- ~~How to add/update **metadata** (in the _Logistics_ step) and at least one practical example~~
- ~~How to use **dynamic templates** and at least one practical example~~
- ~~Available **advanced options** and when to use them~~
- ~~How to add **aliases** and at least one practical example~~

Also, how far back should we backport?

## Checklist

- [x] @colleenmcginnis create initial outline
- [x] @bmorelli25 initial review of the approach + answer the questions above
- [x] @colleenmcginnis address initial feedback
- [ ] @elastic/apm-server review
- [ ] @colleenmcginnis address feedback, merge

## Related issues

Closes https://github.com/elastic/observability-docs/issues/2206

### CODE REVIEW

File: docs/custom-index-template.asciidoc [hunk 1]
```diff
@@ -67,6 +79,34 @@ Specify the number of replica shards in the **index settings**:
...
+
+// [discrete]
+// ==== Aliases
+
+// In the *Aliases* step, ...
+// For example, you could:
+
 Changes to component templates are not applied retroactively to existing indices.
 For changes to take effect, you must create a new write index for the data stream.
 This can be done with the {es} {ref}/indices-rollover-index.html[Rollover API].
```
Comment (bmorelli25) [inline docs/custom-index-template.asciidoc:112]: I think we should add a new heading to this section so it feels more separated from the previous topic.

**Comment by bmorelli25** [comment]
My thoughts:

> How to add/update metadata (in the Logistics step) and at least one practical example

I suppose we could add something here about how you can add arbitrary version numbers and metadata if they help you outside of the product. I'd be more tempted to just add a quick sentence or two and link to this page for more details: https://www.elastic.co/guide/en/elasticsearch/reference/8.11/indices-component-template.html#component-templates-metadata

> How to use dynamic templates and at least one practical example

If we had a good example of how our users might use this feature, then it would be very easy to add something here. 
@elastic/apm-server any recommendations on how someone might use dynamic mappings with APM?

> Available advanced options and when to use them

I don't know enough about this and the potential it does/doesn't have to mess up the APM app if folks touch these settings

> How to add aliases and at least one practical example

Similar response from me here.

cc @axw for the tech lead perspective on these questions

**Comment by axw** [comment]
I can't think of any common use cases related to the APM data streams/index templates for dynamic templates, advanced mapping options, or aliases.

**Comment by simitt** [comment]
>I can't think of any common use cases related to the APM data streams/index templates for dynamic templates, advanced mapping options, or aliases.

++ 
Is there a specific request to add docs for these? Based on previous user questions, I'd say that describing how to customize component templates for adding field mappings and how to change retention periods is the most critical piece (don't remember anyone asking for dynamic templates).

**Review by simitt — APPROVED** [review]

**Pull request merged**